memdx/op_bootstrap.go

```diff
@@ -195,20 +232,33 @@ func (a OpBootstrap) Bootstrap(d Dispatcher, opts *BootstrapOptions, cb func(res
 			currentStage = stageCallback
 			maybeCallback()
 		})
-		dispatchCallback()
+		if err != nil {
+			return err
+		}
+		pendingOp.Add(op)
+
+		return dispatchCallback()
 	}
 
-	dispatchCallback = func() {
+	dispatchCallback = func() error {
 		// this function does nothing because we have no way to easily
 		// schedule a callback to be invoked serialy by the connection
 		// so we rely on maybeCallback() to handle this behaviour.
+		return nil
 	}
 
 	// maybeCallback must be invoked before any dispatches to ensure
 	// we still own all the state and to avoid racing those threads.
 	maybeCallback()
 
-	dispatchHello()
+	if err := dispatchHello(); err != nil {
+		return nil, err
+	}
+
+	return pendingOp, nil
```

Comment:
I wonder if we should have a guard here (or even a comment) that talks about how the pendingOp would be broken if there were no ops added to it, but that the maybeCallback call _should_ prevent that.